.github/workflows/reviewdog.yml

```
@@ -0,0 +1,29 @@
+name: reviewdog
+on:
+  push:
+    branches:
+      - master
+  pull_request:
+
+jobs:
+  golangci-lint:
+    permissions:
+      checks: write
+      contents: read
+      pull-requests: write
+    if: github.event_name == 'pull_request'
+    name: runner / golangci-lint
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          fetch-depth: 0 # required for new-from-rev option in .golangci.yml
+      - name: Run linters & report
+        env:
+          REVIEWDOG_GITHUB_API_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        uses: reviewdog/action-golangci-lint@v2
```

It's safer to pin the action to a specific commit per https://docs.github.com/en/actions/security-guides/security-hardening-for-github-actions#using-third-party-actions.
```suggestion
        uses: reviewdog/action-golangci-lint@70498f2d1f75a55ee9a4d719e74e21ed68aebea3
```